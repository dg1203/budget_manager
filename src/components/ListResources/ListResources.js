import React, { useState } from "react";
import PropTypes from "prop-types";
import {
  List,
  ListItem,
  ListItemAvatar,
  ListItemText,
  ListItemSecondaryAction,
  IconButton,
  Avatar,
  Tooltip,
  Typography
} from "@material-ui/core";
import FolderIcon from "@material-ui/icons/Folder";
import RepeatIcon from "@material-ui/icons/Repeat";
import CompareArrowsIcon from "@material-ui/icons/CompareArrows";
import TransferDialog from "../TransferDialog";
import PaymentDialog from "../PaymnetDialog";

const ListResources = ({ resources, jarId }) => {
  const [openDialog, setOpenDialog] = useState(false);
  const [openPaymentDialog, setOpenPaymentDialog] = useState(false);
  const [choosedCurrency, setChoosedCurrency] = useState("");
  const [choosedAmount, setChoosedAmount] = useState("");
  const openPaymentDialogMethod = (currency, amount) => {
    setChoosedCurrency(currency);
    setChoosedAmount(amount);
    setOpenPaymentDialog(true);
  };
  const openTransferDialog = (currency, amount) => {
    setChoosedCurrency(currency);
    setChoosedAmount(amount);
    setOpenDialog(true);
  };
  return resources.length === 0 ? (
    <Typography variant="h2" component="h2">
      Brak środków w słoiku
    </Typography>
  ) : (
    <>
      <List dense={false}>
        {resources.map((resource, index) => {
          const res = Object.entries(resource)[0];
          const [key, value] = res;
          return (
            <ListItem key={index}>
              <ListItemAvatar>
                <Avatar>
                  <FolderIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText primary={`${key} ${value}`} />
              <ListItemSecondaryAction>
                <Tooltip
                  title="Przenieś środki do innego słoika"
                  placement="top"
                >
                  <IconButton onClick={() => openTransferDialog(key, value)}>
                    <CompareArrowsIcon />
                  </IconButton>
                </Tooltip>
                <Tooltip title="Wyciągnij środki" placement="top">
                  <IconButton
                    onClick={() => openPaymentDialogMethod(key, value)}
                  >
                    <RepeatIcon />
                  </IconButton>
                </Tooltip>
              </ListItemSecondaryAction>
            </ListItem>
          );
        })}
      </List>
      <TransferDialog
        open={openDialog}
        openDialog={setOpenDialog}
        currency={choosedCurrency}
        amount={choosedAmount}
        jarId={jarId}
      />
      <PaymentDialog
        open={openPaymentDialog}
        openDialog={setOpenPaymentDialog}
        jarId={jarId}
        currency={choosedCurrency}
        amount={choosedAmount}
      />
    </>
  );
};

ListResources.propTypes = {
  resources: PropTypes.array.isRequired,
  jarId: PropTypes.string.isRequired
};

export default ListResources;
