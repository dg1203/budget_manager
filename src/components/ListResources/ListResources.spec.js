import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import ListResources from "./ListResources";
import store from "../../store";

const defaultProps = {
  jarId: "dg3hgaj",
  resources: [
    {
      PLN: 200,
      USD: 350,
      GBP: 425
    }
  ]
};

const renderComponent = (props) =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <ListResources {...props} />
      </MemoryRouter>
    </Provider>
  );

describe("Header component", () => {
  it("renders the component", () => {
    const container = renderComponent(defaultProps);
    expect(container).toMatchSnapshot();
  });
  it("information about empty resources list should't be render", () => {
    const { queryByTestId } = renderComponent(defaultProps);
    expect(queryByTestId("info-emptyresources")).toBeFalsy();
  });
  it("information about empty resources list should be render if resources list is empty", () => {
    const { getByTestId } = renderComponent({
      ...defaultProps,
      resources: []
    });
    expect(getByTestId("info-emptyresources")).toBeTruthy();
  });
  it("test open transfer dialog", () => {
    const { getByTestId,  } = renderComponent(defaultProps);
    fireEvent.click(getByTestId("open-transferdialog"));
    expect(getByTestId("transfer-dialog")).toBeTruthy();
  });
  it("test open payment dialog", () => {
    const { getByTestId } = renderComponent(defaultProps);
    fireEvent.click(getByTestId("open-paymentdialog"));
    expect(getByTestId("payment-dialog")).toBeTruthy();
  });
});
