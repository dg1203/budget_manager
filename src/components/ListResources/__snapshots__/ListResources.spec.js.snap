// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root MuiList-padding"
      >
        <li
          class="MuiListItem-container"
        >
          <div
            class="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
          >
            <div
              class="MuiListItemAvatar-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                PLN 200
              </span>
            </div>
          </div>
          <div
            class="MuiListItemSecondaryAction-root"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="open-transferdialog"
              tabindex="0"
              title="Przenieś środki do innego słoika"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9.01 14H2v2h7.01v3L13 15l-3.99-4v3zm5.98-1v-3H22V8h-7.01V5L11 9l3.99 4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="open-paymentdialog"
              tabindex="0"
              title="Wyciągnij środki"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root MuiList-padding"
    >
      <li
        class="MuiListItem-container"
      >
        <div
          class="MuiListItem-root MuiListItem-gutters MuiListItem-secondaryAction"
        >
          <div
            class="MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              PLN 200
            </span>
          </div>
        </div>
        <div
          class="MuiListItemSecondaryAction-root"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="open-transferdialog"
            tabindex="0"
            title="Przenieś środki do innego słoika"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.01 14H2v2h7.01v3L13 15l-3.99-4v3zm5.98-1v-3H22V8h-7.01V5L11 9l3.99 4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="open-paymentdialog"
            tabindex="0"
            title="Wyciągnij środki"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
